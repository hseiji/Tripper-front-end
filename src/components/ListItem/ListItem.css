
.card-item {
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  transition: 0.3s;
  margin: 5px;
  background-color: lightcyan;
  padding: 10px;
}

.card-item:hover {
  box-shadow: 0 8px 16px 0 rgba(0,0,0,0.3);
}
