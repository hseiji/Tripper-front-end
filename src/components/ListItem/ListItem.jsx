import React, { useContext } from "react";
import { AppContext } from "../hooks/useAppContext";
import './ListItem.css';

export const ListItem = (props) => {
  const { addToMap, selectedPlan } = useContext(AppContext);

  // Add marker
  const addMarker = () => {
    addToMap({
      id: `${props.results.id}plan${selectedPlan}`,
      name: props.results.name,
      lat: props.results.latitude,
      lng: props.results.longitude,
      done: false,
      alias: props.results.alias,
      url: props.results.url,
      address: props.results.address,
      image_url: props.results.image_url,
    });
    console.log("adding event", props.results);
  };

  return (
    <li>
      {props && (
        <div className="card-item">
          <a href={props.results.url} target="_blank" rel="noopener noreferrer" style={{"fontSize":"20px", "fontWeight":"300"}}>{props.results.name}</a>
          <p style={{"fontSize":"15px"}}>{props.results.address}</p>
          <p style={{"fontSize":"15px"}}>
            Rating: {props.results.rating} ({props.results.review} reviews)
          </p>
          <a href={props.results.url} target="_blank" rel="noopener noreferrer">
            <img
              src={props.results.image_url}
              alt=""
              height="150px"
              width="150px"
            />
          </a>
          {/* <button onClick={addMarker}>Add</button> */}
          {/* <i className="fa-regular fa-location-plus" onClick={addMarker}></i> */}
          <i className="fa-regular fa-2x fa-square-plus" onClick={addMarker}></i>
        </div>
      )}
    </li>
  );
};
