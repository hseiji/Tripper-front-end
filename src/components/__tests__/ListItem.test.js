import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { ListItem } from '../ListItem/ListItem';

afterEach(() => {
  cleanup();
})

test('componenet should be present in document and have text', () => {
  const results = [
    {
      id: 1,
      name: "Saku Sushi",
      url: "https://www.yelp.com/biz/saku-sushi-toronto-2?adjust_creative=zGbNrmtfS5j0G5IKmn8ufA&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_search&utm_source=zGbNrmtfS5j0G5IKmn8ufA",
      address: "",
      review: "",
      rating: "4",
      image_url: "https://s3-media3.fl.yelpcdn.com/bphoto/xBtbkhP7VY6x6kTuKlznxg/o.jpg",
    }
  ]
  render(<ListItem results={results}/>);
  const myElement = screen.getByTestId('list-item-id');
  expect(myElement).toBeInTheDocument();
  expect(myElement).toHaveTextContent("Rating");
});

test('matches snapshot', () => {
  const results = [
    {
      id: 1,
      name: "Saku Sushi",
      url: "https://www.yelp.com/biz/saku-sushi-toronto-2?adjust_creative=zGbNrmtfS5j0G5IKmn8ufA&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_search&utm_source=zGbNrmtfS5j0G5IKmn8ufA",
      address: "",
      review: "",
      rating: "4",
      image_url: "https://s3-media3.fl.yelpcdn.com/bphoto/xBtbkhP7VY6x6kTuKlznxg/o.jpg",
    }
  ];
  const tree = renderer.create(<ListItem results={results}/>).toJSON();
  expect(tree).toMatchSnapshot(); 
});