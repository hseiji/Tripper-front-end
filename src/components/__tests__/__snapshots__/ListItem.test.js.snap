// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<li
  data-testid="list-item-id"
>
  <div
    className="card-item"
  >
    <a
      rel="noopener noreferrer"
      style={
        {
          "fontSize": "20px",
          "fontWeight": "300",
        }
      }
      target="_blank"
    />
    <p
      style={
        {
          "fontSize": "15px",
        }
      }
    />
    <p
      style={
        {
          "fontSize": "15px",
        }
      }
    >
      Rating: 
       (
       reviews)
    </p>
    <a
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt=""
        height="150px"
        width="150px"
      />
    </a>
    <i
      className="fa-regular fa-2x fa-square-plus"
      onClick={[Function]}
    />
  </div>
</li>
`;
